java-checks/src/main/java/org/sonar/java/checks/AssertionsInTestsCheck.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012 SonarSource
+ * dev@sonar.codehaus.org
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.Lists;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.checks.methods.MethodInvocationMatcher;
+import org.sonar.java.checks.methods.NameCriteria;
+import org.sonar.java.checks.methods.TypeCriteria;
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.AnnotationTree;
+import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+
+@Rule(
+  key = "S2699",
+  name = "Tests should include assertions",
+  tags = {"junit"},
+  priority = Priority.CRITICAL)
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.UNIT_TESTABILITY)
+@SqaleConstantRemediation("10min")
+public class AssertionsInTestsCheck extends BaseTreeVisitor implements JavaFileScanner {
+
+  private static final String GENERIC_ASSERT = "org.fest.assertions.GenericAssert";
+  private static final MethodInvocationMatcher FEST_AS_METHOD = MethodInvocationMatcher.create()
+    .typeDefinition(GENERIC_ASSERT).name("as").withNoParameterConstraint();
+  private static final MethodInvocationMatcher FEST_DESCRIBED_AS_METHOD = MethodInvocationMatcher.create()
+    .typeDefinition(GENERIC_ASSERT).name("describedAs").withNoParameterConstraint();
+  private static final MethodInvocationMatcher FEST_OVERRIDE_ERROR_METHOD = MethodInvocationMatcher.create()
+    .typeDefinition(GENERIC_ASSERT).name("overridingErrorMessage").withNoParameterConstraint();
+  private static final ArrayList<MethodInvocationMatcher> ASSERTION_INVOCATION_MATCHERS = Lists.newArrayList(
```

Comment:
:new: ![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)The type of the "ASSERTION_INVOCATION_MATCHERS" object should be an interface such as "List" rather than the implementation "ArrayList".[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)